srcts/src/shiny/render.ts

```diff
@@ -209,9 +213,28 @@ function renderDependency(dep_: HtmlDep) {
     $head.append(stylesheetLinks);
   }
 
+  const scriptPromises: Array<Promise<any>> = [];
+  const scriptElements: HTMLScriptElement[] = [];
+
   dep.script.forEach((x) => {
     const script = document.createElement("script");
 
+    if (!x.hasOwnProperty("async")) {
+      // Set async to false by default, so that if there are multiple script
+      // tags, they are guaranteed to run in order. For dynamically added
+      // <script> tags, browsers set async to true by default, which differs
+      // from static <script> tags in the html, which default to false.
+      //
+      // Refs:
+      // https://stackoverflow.com/a/8996894/412655
+      // https://jason-ge.medium.com/dynamically-load-javascript-files-in-order-5318ac6bcc61
+      //
+      // Note that one odd thing about these dynamically-created <script> tags
+      // is that even though the JS object's `x.script` property is true, it
+      // does NOT show up as a property on the <script> element.
```

Comment:
For future reference, this is what happens when I run this in Chrome. The dynamic `script` tag has `script.async` set to `true`, but it doesn't show it on the DOM element, unless it's explicitly set.


```R
let script = document.createElement('script');
#> <script>​</script>​

script.async
#> true
script
#> <script>​</script>​

script.async = true
script
#> <script async>​</script>​
```

Also, setting it to `false` makes no visible change on the DOM element.

```R
let script = document.createElement('script');
#> <script>​</script>​

script.async
#> true
script
#> <script>​</script>​

script.async = false
script.async
#> false
script
#> <script>​</script>
```